### DISCUSSION TITLE
On-/offline-settings using MSM inconsistent and malfunctioning.

### DISCUSSION THREAD

**Discussion by silenz88** [ExpressionEngine/ExpressionEngine#3043] [Feature Requests]
**Description of the problem**

EE 7.2.7
MSM 

This one is kind of confusing so I hope I will be able to explain it.

Given is an installation with Multiple Site Manager enabled and 6 sites.

There a two "Website online?" switches  which interact inconsistently and sometimes even act across multiple sites.

1.) in System Settings / General Settings (/cp/settings/general)

2.) under MSM (/cp/msm/edit)

Also under MSM (/cp/msm) there is the list of sites that allegedly displays the current online status, which it doesn't neccessarily. 
![msm](https://user-images.githubusercontent.com/6132998/221158490-1d3f0f31-e790-49c4-b588-5c513ed86052.jpg)

In the screenshot site #1 is online.
When I go to General Settings and switch the Website to offline the site still is shown as Online in this list while it is in fact offline.

At least UX-wise this is misleading.

When I switch to site #2 and look under General Settings the switch says the website is online. But it is not since the switch under MSM is set to offline. So the MSM-switch overrides the General Settings-switch. You have to activate both to set the site online. I don't think this is reflected in the documentation.

Really problematic is the switch under General Settings. I'd consider it a bug.

The docs state:

> This section of the Control Panel allows you to define the basic settings of your site. If you are using [Multiple Sites](https://docs.expressionengine.com/latest/msm/overview.html), note that these settings are per-site.

**If you flip the switch under General Settings it affects all sites.** 

So if you are in site #3 and set the website to offline in General Settings all websites go offline not just #3. Likewise if you switch any site to online here all sites are set to online.

So the proper way to operate under MSM would be to always have alls sites switched on under General Settings and exclusively use the MSM-switches to control the status.

_This needs to be at least documented but preferably foolproofed otherwise it leaves you with people taking multiple sites offline without noticing._

By either synchronizing the two switches or by making the switch under General Settings site-aware or get rid of it when using MSM.

On a side-note: The only way I found to reach the MSM site-overview (/cp/msm) ist by going to "+Add Site" and then clicking on "Sites" in the breadcrumb-navigation.
![msm4](https://user-images.githubusercontent.com/6132998/221168384-60152322-3214-4228-b836-f2dec90e5009.png)

Not really intuitive. Is there any other way?

**Comment by intoeetive** [comment]
@silenz88 there is indeed confusion here. ExpressionEngine, when used with MSM, has two separate settings:
`is_system_on` - global preference
`is_site_on` - per-site preference

Both are labelled the same however (but found on different settings pages).

I think we need to make following additions:
- add indication if system is offline, either globally or on sites listing page
- the settings need to be named differently to avoid confusion
- update the documentation to bring more clarity

This is not bug in terms on system functioning, but I agree it's a UX / documentation issue that we need to work on (converting this to discussion so people could contribute UX ideas)

As of your side note, when MSM is on, there's `Sites` link in the developer menu in sidebar.

**Comment by intoeetive** [comment]
The pull request has been opened to bring extra clarity on these settings: https://github.com/ExpressionEngine/ExpressionEngine/pull/3739

It leaves the label and description for setting as it is when there's only one site in the system and is changing it if MSM is enabled.
